support-lib/jni/Marshal.hpp

```diff
@@ -80,7 +80,11 @@ namespace djinni
 		Bool() : Primitive("java/lang/Boolean", "valueOf", "(Z)Ljava/lang/Boolean;", "booleanValue", "()Z") {}
 		friend JniClass<Bool>;
 		friend Primitive<Bool, bool, jboolean>;
-		static JniType unbox(JNIEnv* jniEnv, jmethodID method, jobject j) noexcept { return jniEnv->CallBooleanMethod(j, method); }
+		static JniType unbox(JNIEnv* jniEnv, jmethodID method, jobject j) noexcept {
+            const auto result = jniEnv->CallBooleanMethod(j, method);
+            jniExceptionCheck(jniEnv);
```

Comment:
I agree.  A jniCallBooleanMethod(env, obj, args...) wrapper would make this sort of case much less error prone, and perfect-forwarding could make it zero-cost.  I don't want to deal with creating that and modifying all the code to use it at the moment, but I'd welcome it at some point.